hdfs-commons/src/main/java/org/apache/mesos/hdfs/config/HdfsFrameworkConfig.java

```diff
@@ -61,6 +61,23 @@ public HdfsFrameworkConfig() {
     setConf(configuration);
   }
 
+  public String getPrincipal() {
+    return getConf().get("mesos.hdfs.principal", "");
+  }
+
+  public String getSecret() {
+    return getConf().get("mesos.hdfs.secret", "");
+  }
+
+  public boolean cramCredentialsEnabled() {
+    String principal = getPrincipal();
+    String secret = getSecret();
```

Comment:
In Java, you would need to first check for null before checking the string value. Also, it would need to be !principal.equals("") or preferably !principal.isEmpty() otherwise it will not actually do the string comparison.

```principal != null && !principal.isEmpty()```